internal/core/application/wallet_service.go

```diff
@@ -752,18 +757,26 @@ func (w *walletService) getAllUnspentsForAccount(
 type unspentInfo struct {
 	info     domain.AddressInfo
 	unspents []domain.Unspent
+	err      error
 }
 
 func (w *walletService) restoreUnspents(
 	info domain.AddressesInfo,
 	chRes chan *InitWalletReply,
 ) ([]domain.Unspent, error) {
 	chUnspentsInfo := make(chan unspentInfo)
-	chErr := make(chan error, 1)
 	unspents := make([]domain.Unspent, 0)
 	cb := newCircuitBreaker()
+	wg := &sync.WaitGroup{}
+	wg.Add(len(info))
 
-	for _, in := range info {
+	go func() {
+		wg.Wait()
+		close(chUnspentsInfo)
+	}()
+
+	for i := range info {
```

Comment:
I dont think we need waitgroups in here

```suggestion
	go func() {
	    defer close(chUnspentsInfo)
	    for i := range info {
		in := info[i]
		path := strings.Split(in.DerivationPath, "/")
		addressIndex, _ := strconv.Atoi(path[len(path)-1])
		chRes <- &InitWalletReply{
			AccountIndex: in.AccountIndex,
			AddressIndex: addressIndex,
			Status:       Processing,
			Data: fmt.Sprintf(
				"account %d index %d", in.AccountIndex, addressIndex,
			),
		}
		go w.restoreUnspentsForAddress(cb, in, chUnspentsInfo, wg)
		time.Sleep(1 * time.Millisecond)
	}
	}
```